crates/router/src/core/relay/utils.rs

```diff
@@ -0,0 +1,139 @@
+use std::str::FromStr;
+
+use common_utils::{ext_traits::OptionExt, id_type};
+use error_stack::ResultExt;
+use hyperswitch_domain_models::{router_data::ErrorResponse, types};
+
+use crate::{
+    core::payments,
+    db::{
+        domain,
+        errors::{self, RouterResult},
+    },
+    routes::SessionState,
+};
+
+const IRRELEVANT_PAYMENT_INTENT_ID: &str = "irrelevant_payment_intent_id";
+
+const IRRELEVANT_PAYMENT_ATTEMPT_ID: &str = "irrelevant_payment_attempt_id";
+
+pub async fn construct_relay_refund_router_data<'a, F>(
+    state: &'a SessionState,
+    connector_name: &str,
+    merchant_id: &id_type::MerchantId,
+    connector_account: &domain::MerchantConnectorAccount,
+    relay_record: &hyperswitch_domain_models::relay::Relay,
+) -> RouterResult<types::RefundsRouterData<F>> {
+    let connector_auth_type = connector_account
+        .get_connector_account_details()
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Failed while parsing value for ConnectorAuthType")?;
+
+    let webhook_url = Some(payments::helpers::create_webhook_url(
+        &state.base_url.clone(),
+        merchant_id,
+        connector_name,
+    ));
```

Comment:
We should consider using merchant connector account ID instead of connector name.
We can take this change up in a separate PR.